src/main/java/com/ibm/crail/storage/blkdev/BlkDevStorageClient.java

```diff
@@ -8,22 +8,87 @@
 import com.ibm.crail.storage.blkdev.client.BlkDevStorageEndpoint;
 
 import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.StringTokenizer;
+import java.util.HashMap;
+import java.util.List;
 import org.slf4j.Logger;
 
 public class BlkDevStorageClient implements StorageClient{
     private static final Logger LOG = CrailUtils.getLogger();
+	private HashMap<Long, String> nodeMap;
 
 	public void printConf(Logger logger) {
-		BlkDevStorageConstants.printConf(logger);
+		BlkDevStorageConstants.printClientConf(logger);
 	}
 
+	private long convertIPPortToKey(String ipPort) throws IOException {
+		//The first token is the ip and the second is the port
+		StringTokenizer tokenizer = new StringTokenizer(ipPort, ":");
+		String ip = tokenizer.nextToken();
+		byte[] bytes = InetAddress.getByName(ip).getAddress();
+		int port = Integer.parseInt(tokenizer.nextToken());
+
+		return DataNodeInfo.calcKey(bytes, port);	

```

Comment:
They key calculation method might change in the future and I'm not sure exporting calcKey like this is the right thing to do so I recommend using getIP/getPort on the DataNodeInfo and create your own hashcode method.

Reply:
No problem. It'll be a duplicate of the current key() implementation for now.